src/main/java/service/imp/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,28 @@
+package service.imp;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import java.util.Map;
+import java.util.Set;
+import service.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String COLUMNS_NAME = "fruit,quantity";
+    private FruitDao fruitDao;
```

Comment:
The `fruitDao` should be private final if it's not meant to be changed after initialization. Also, consider using dependency injection for `FruitDao` instead of instantiating it directly in the constructor to follow the SOLID principles, particularly the Dependency Inversion Principle.